packages/dd-trace/src/encode/0.4.js

```diff
@@ -263,6 +271,45 @@ class AgentEncoder {
     }
   }
 
+  _encodeObject (bytes, value, circularReferencesDetector = new Set()) {
+    circularReferencesDetector.add(value)
+    if (Array.isArray(value)) {
+      return this._encodeObjectAsArray(bytes, value, circularReferencesDetector)
+    } else if (typeof value === 'object') {
+      return this._encodeObjectAsMap(bytes, value, circularReferencesDetector)
+    } else if (typeof value === 'string' || typeof value === 'number') {
+      this._encodeValue(bytes, value)
+    }
+  }
+
+  _encodeObjectAsMap (bytes, value, circularReferencesDetector) {
+    const keys = Object.keys(value)
+    const validKeys = keys.filter(
+      key =>
+        typeof value[key] === 'string' || typeof value[key] === 'number' ||
+        (value[key] !== null && typeof value[key] === 'object' && !circularReferencesDetector.has(value[key])))
```

Comment:
what the f is this weird syntax ? absolutely unreadable. Was it like that in the file before ?

Reply:
maybe something like this ?
```suggestion
    const validKeys = keys.filter(key => {
      return typeof value[key] === 'string'
        || typeof value[key] === 'number'
        || (value[key] !== null && typeof value[key] === 'object' && !circularReferencesDetector.has(value[key]))
    })
```